anchore_engine/services/policy_engine/engine/feeds/client.py

```diff
@@ -247,7 +423,87 @@ def _extract_response_data(self, response_text):
         return next_token, response_text, count
 
 
-def get_client(
+class GrypeDBServiceClient(IFeedSource):
+    RETRY_COUNT = 3
+
+    def __init__(
+        self,
+        grype_db_endpoint: str,
+        http_client: HTTPBasicAuthClient,
+    ):
+        self.feed_url = grype_db_endpoint
+        self.http_client = http_client
+
+    def list_feeds(self) -> FeedList:
+        return FeedList(
+            feeds=[
+                FeedAPIRecord(
+                    name="grypedb",
+                    description="grypedb feed",
+                    access_tier="0",
+                )
+            ]
+        )
+
+    def list_feed_groups(self, feed: str) -> FeedGroupList:
+        return FeedGroupList(
+            groups=[
+                FeedAPIGroupRecord(
+                    name="grypedb:vulnerabilities",
+                    description="grypedb:vulnerabilities group",
+                    access_tier="0",
+                )
+            ]
+        )
+
```

Comment:
Isn't the goal to have feed and feed group information drawn from Grype itself? @zhill brought that up during a short discussion. The policy engine endpoints for feed and feed group information relies on FeedMetadata and FeedGroupMetadata DB rows populated in anchore_engine.services.policy_engine.engine.feeds.sync.DataFeeds.sync_metadata(). That method calls these two methods that we're talking about here, anchore_engine.services.policy_engine.engine.feeds.client.GrypeDBServiceClient.list_feeds() and list_feed_groups(). I'm not sure how to implement this without breaking the legacy flow.

Reply:
We could make the api controller at anchore_engine.services.policy_engine.api.controllers.feeds.list_feeds() read the config and generate FeedGroupMetadata and FeedMetadata records using the grype wrapper, and then glue those records together with the legacy ones in the database. This would let us surface grype info without duplicating data from Grype DB in Anchore DB feeds tables.